### ISSUE TITLE
Elision locations moving in export after footnotes

### ISSUE THREAD

**Issue by bjohnson747** [harvard-lil/h2o#775]
Exported case with annotations has its elisions not covering desired elided text / elisions are moving. Unsure if what tweak this is due to.

Test case: https://opencasebook.org/casebooks/128853-constitutional-law-for-lessig/resources/4.2.7.2-board-v-garrett/annotate

On H2O:

<img width="968" alt="Screen Shot 2019-05-03 at 3 43 29 PM" src="https://user-images.githubusercontent.com/12850389/57161831-85f0b480-6dba-11e9-8ad7-8f66d4792f46.png">

Moved elisions on export:

![Screen Shot 2019-05-03 at 3 43 32 PM](https://user-images.githubusercontent.com/12850389/57161842-8ab56880-6dba-11e9-93f4-41940cc39afb.png)

Occurring in other spots of this case as well

**Comment by bjohnson747** [comment]
Link to the HTML of the example case: https://opencasebook.org/admin/case/8356/edit

**Comment by bjohnson747** [comment]
(tagging @leppert @rebeccacremona in case relevant to your foci)

**Comment by cgruppioni** [comment]
Looking into this

**Comment by leppert** [comment]
Not sure if this is related, but I recently found that, in some instances, the counts were different between nokogiri and the front end due to nokogiri adding \r alongside \n

https://github.com/leppert/h2o/commit/734bcb936eabb09cba0439f0a19d51fd93bbf0c1

**Comment by cgruppioni** [comment]
@leppert This would be an issue that has been happening for awhile then, right? Now a recent change

**Comment by leppert** [comment]
Correct, though it’s unclear to me how well tested the exports were given that fact; not sure what’s a new bug and what’s a bug that’s newly discovered.

**Comment by bjohnson747** [comment]
Casebooks that don't have cases tagged in the specific way the batch of cases being used in the example above - with page numbers and footnotes tagged in a specific fashion - seem to be fine- ex. the elisions in this exported are fine: https://opencasebook.org/casebooks/25965-contracts/sections/1-part-i-principles-of-contractual-obligation 

Digging into this some more to drill down

**Comment by bjohnson747** [comment]
Successfully drilled down - issue seems to start after a footnote text has appeared, and it's moving elisions up.

In this example, all the elisions before the screenshot paragraph are located fine in the export. Once I hit this asterisk-ed footnote, the elisions that follow it are being placed improperly on export.

H2O:

<img width="1035" alt="Screen Shot 2019-05-06 at 12 36 12 PM" src="https://user-images.githubusercontent.com/12850389/57240290-0c480900-6ffc-11e9-87c4-e3363ccc71c0.png">

Export:

![Screen Shot 2019-05-06 at 12 36 33 PM](https://user-images.githubusercontent.com/12850389/57240298-10742680-6ffc-11e9-9ec8-926aec865abb.png)


This is from: https://opencasebook.org/casebooks/79649-sam-contracts-for-testing/resources/2.1.5-griswold-v-connecticut/annotate

**Comment by leppert** [comment]
Ah, yes, ignore my previous comment then. This appears to be shifting across entire paragraphs.

**Comment by cgruppioni** [comment]
Thank you @bjohnson747! There have been no recent changes to export / export footnotes so this must have been happening for at least a few months... I'll dig into it more first thing tomorrow

**Comment by bjohnson747** [comment]
@cgruppioni yeah, it's something to do with footnotes from CAP cases - either manually tweaked by us, or not - throwing any annotations that come after of them off.

For example, I just added this case from CAP a minute ago, and it's got the same issue: https://opencasebook.org/casebooks/128853-constitutional-law-for-lessig/resources/4.3.2.4-pga-tour-inc-v-martin/annotate
Could the functionality that you fixed and/or the adjusted formatting for footnotes on the live site have had an effect on the export for some reason?

**Comment by cgruppioni** [comment]
@bjohnson747 No the code for the web footnotes is separate from the export footnotes. The ticket I'm working on https://github.com/harvard-lil/h2o/issues/677 could fix this on it's own

**Comment by leppert** [comment]
@cgruppioni @bjohnson747 It looks like this is a result of `aside` being removed from the paragraph filter in the Ruby and JS:
https://github.com/harvard-lil/h2o/commit/9626cb48f412e56f5f4d2b99d25945d645208fd2

And then restored to the Ruby but not the JS:
https://github.com/harvard-lil/h2o/commit/504b5e5d256c154ef50837b48fc69aa6956a9d91

Once the Ruby hits `aside` tags, the paragraph counts start to drift away from what the JS counted and was subsequently stored in the DB.

**Comment by cgruppioni** [comment]
Thanks @leppert

**Comment by bjohnson747** [comment]
For further testing, I tweaked a private case to have just <p> tags, with one of its paragraph also encased in <aside> tags too. There are no additional tags (ex. anything footnote-related).
The 7th 'paragraph' in this case, starting with "Wireless carriers collect and store" is the one that also has <aside> tags. https://opencasebook.org/admin/case/8044/edit

When I exported this, the elisions in that one paragraph all moved forward a single character, and elisions that were in regular <p> tagged paragraphs are in the correct spot. 

This is the item i was testing on: https://opencasebook.org/casebooks/73057-zittrain-torts-playlist-rename/resources/1.1.4-carpenters-v-united-states/annotate
